source/mesh_refinement/isotherms.cc

```diff
@@ -0,0 +1,373 @@
+                             "four values separted by a comma. The first two values indicate the minimum and maximum "
+                             "refinement levels respecitvely. The can be an integer, but also the words min or max "
+                             "optionally followed by a plus or minus a number. For example 'min+2' or 'max-1'. The "
+                             "last two values are the mimum and maximum temperature respectively between which the "
+                             "mimumun and maximum refinement are enforced."
+                            );
+
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+    template <int dim>
+    void
+    Isotherms<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      std::vector<std::pair<std::string,std::string> > isotherms_raw;
+      prm.enter_subsection("Mesh refinement");
+      {
+        minimum_refinement_level = Utilities::string_to_int(prm.get ("Minimum refinement level"));
+        maximum_refinement_level = Utilities::string_to_int(prm.get("Initial global refinement")) + Utilities::string_to_int(prm.get("Initial adaptive refinement"));
+        prm.enter_subsection("Isotherms");
+        {
+          exclude_composition = Utilities::string_to_int(prm.get("Exclude composition"));
+          std::vector<std::string> isotherms_outer_loop = Utilities::split_string_list(prm.get ("List of isotherms"),';');
+          // process the List of isotherms to get all data out of the structure
+          isotherms_raw.resize(isotherms_outer_loop.size());
+          isotherms.resize(isotherms_outer_loop.size());
+          // loop through all given isotherms
+          for ( unsigned int isotherm_line = 0; isotherm_line != isotherms_raw.size(); ++isotherm_line )
+            {
+              std::vector<std::string> isotherms_inner_loop = Utilities::split_string_list(isotherms_outer_loop[isotherm_line]);
+
+              // The minimum and maximum refinemnt levels respecively
+              isotherms_raw[isotherm_line].first  = isotherms_inner_loop[0];
+              isotherms_raw[isotherm_line].second = isotherms_inner_loop[1];
+              // The temperatures for the min and max isotherms respectively
+              isotherms[isotherm_line].first  = Utilities::string_to_double(isotherms_inner_loop[2]);
+              isotherms[isotherm_line].second = Utilities::string_to_double(isotherms_inner_loop[3]);
+            }
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+
+      isotherms_levels.resize(isotherms_raw.size());
+      // convert max,max-1,max-2 etc and min, min-1,min-2 to numbers in integer
+      for ( unsigned int isotherm_line = 0; isotherm_line != isotherms_levels.size(); ++isotherm_line )
+        {
+          if (isotherms_raw[isotherm_line].first.compare(0,3,"max") == 0)
+            {
+              if (isotherms_raw[isotherm_line].first.compare(0,4,"max-") == 0)
```

Comment:
here and below, you are assuming that a user actually writes `max-2`, but not `max - 2` or similar. That's ok, but I think you need to do some error checking here. For example, the if you entered `max - 2`, you'd fall into the `else` branch below where you just assume that the user had written `max`.